app/controllers/twilio_webhooks_controller.rb

```diff
@@ -38,12 +38,17 @@ def create_incoming_text_message
       )
 
       if FileTypeAllowedValidator::VALID_MIME_TYPES.include? content_type
-        extension = MIME::Types[content_type].first.extensions.first
-        filename_with_extension = "#{filename}.#{extension}"
-        document.upload.attach(io: StringIO.new(Net::HTTP.get(URI(media_url))),
-                                          filename: filename_with_extension,
-                                          content_type: content_type,
-                                          identify: false)
+        response = Net::HTTP.get_response(URI(media_url)) # first we get a redirect from Twilio to S3
```

Comment:
I worked with @bytheway875 on attachments earlier, and as far as I recall, we forgot to validate on demo that this worked properly. Therefore, the code on origin/master never tested with actual Twilio responses, since as I recall, we didn't set up a Twilio ngrok for local testing. I think we were basing our implementation on what was in Twilio's docs, but we might have strayed from them by accident.

The above is merely an attempt to describe what happened, not to pass judgement on it. Hope that helps.